src/Stateless/Enforce.cs

```diff
@@ -15,5 +14,35 @@ public static T ArgumentNotNull<T>(T argument, string description)
 
             return argument;
         }
+        public static void NotIdentityTransition<TState, TTrigger>(StateMachine<TState, TTrigger>.StateRepresentation _representation, TState destination)
```

Comment:
I'd rather not go down the path of moving all validation to `Enforce` - it reads nicely at the call site, but it'll lead to a jumble of unrelated things here. `StateConfiguration` still seems like the right home for these.